src/Connection/Resource/Stream.php

```diff
@@ -211,6 +211,18 @@ public function write(string $string): int
             throw new RuntimeException('Unable to write to stream', 1);
         }
 
+        if ($result === 0) {
```

Comment:
I think there's no point to introduce a separate check. Currently the issue is that if statement above is to strict about type. Something like this would fix an issue:

`if ($result == false)`

Reply:
`$result` contains the number of bytes successfully written to the socket buffer. A value of `0` is a **valid** and **successful** result (different from `false`). It typically indicates that the remote side is busy and the local send buffer is full, preventing additional data from being written at the moment.